packages/core/src/Page/PageConfig.ts

```diff
@@ -74,7 +74,7 @@ export class PageConfig {
     baseUrlMap: Set<string>;
     dev: boolean;
     faviconUrl?: string;
-    frontmatterOverride?: { [frontmatterName: string]: string },
+    frontmatterOverride?: Record<string, string>,
```

Comment:
While I do agree that `Record<K,V>` is a cleaner way to represent key-value types, I'm not sure if simply substituting `{ [key: K]: V }` with `Record<K,V>` would be more readable. 

In this case, I would actually prefer the previous type declaration because I would know that the key is actually `frontmatterName`, whereas if I saw it as `Record<string, string>`, I would not know what the meaning of the key is other than the fact that it is a `string`. Wdyt?

Reply:
> In this case, I would actually prefer the previous type declaration because I would know that the key is actually `frontmatterName`, whereas if I saw it as `Record<string, string>`, I would not know what the meaning of the key is other than the fact that it is a `string`. Wdyt?

I agree with this point on the semantic value of the key!
It feels that most of the time the key is `id:` or `name:` which can be omitted without losing information, but in cases like this the key conveys information that may be valuable to the developer. What do you think of leaving a comment in such cases? e.g.
```typescript
{
	frontmatterOverride?: Record<string, string>,  // key: frontmatter name
}
```

Reply:
No problem, just concerns about consistency between the two declarations. In that case should we leave it as such in these cases?